### PR TITLE
{get=>find}Resource in some tests

### LINKED ISSUE #12677: `ZipArchiveTest` and `ZipAndJarFileLookupFactoryTest` fail under _some_ circumstances
## Reproduction steps

Scala version: `2.13.10`, also latest `2.13.x -> 8fa1e7c1`
Sbt version: `1.7.1`
Java `8` and `11`
Maybe you need `java-atk-wrapper.jar` or similar on your classpath, see below.

```scala
sbt test
```
or
```scala
sbt testOnly scala.reflect.io.ZipArchiveTest scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest
```
or
```scala
sbt junit / Test / testOnly scala.reflect.io.ZipArchiveTest scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest
```
or
in Intellij: open respective file, and run via **Debug**. (not Run!)

Output:
```scala
sbt:root> testOnly scala.reflect.io.ZipArchiveTest scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest
...
[info] Test run scala.reflect.io.ZipArchiveTest started
...
[info] Test scala.reflect.io.ZipArchiveTest.manifest resources just works started
[error] Test scala.reflect.io.ZipArchiveTest.manifest resources just works failed: java.lang.AssertionError: null, took 0.028 sec
[error]     at scala.reflect.io.ZipArchiveTest.$anonfun$manifest resources just works$2(ZipArchiveTest.scala:60)
[error]     at scala.reflect.io.ZipArchiveTest.$anonfun$manifest resources just works$2$adapted(ZipArchiveTest.scala:58)
[error]     at scala.util.Using$.$anonfun$resources$2(Using.scala:298)
[error]     at scala.util.Using$.resource(Using.scala:261)
[error]     at scala.util.Using$.$anonfun$resources$1(Using.scala:297)
[error]     at scala.util.Using$.resource(Using.scala:261)
[error]     at scala.util.Using$.resources(Using.scala:296)
[error]     at scala.reflect.io.ZipArchiveTest.manifest resources just works(ZipArchiveTest.scala:58)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]     at java.lang.reflect.Method.invoke(Method.java:566)
[error]     ...
[info] Test run scala.reflect.io.ZipArchiveTest finished: 1 failed, 0 ignored, 5 total, 0.093s
[info] Test run scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest started
[info] Test scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.cacheInvalidation started
[info] Test scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.manifest classpath entry works started
[error] Test scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.manifest classpath entry works failed: java.lang.AssertionError: null, took 0.024 sec
[error]     at scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.$anonfun$manifest classpath entry works$5(ZipAndJarFileLookupFactoryTest.scala:91)
[error]     at scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.$anonfun$manifest classpath entry works$5$adapted(ZipAndJarFileLookupFactoryTest.scala:88)
[error]     at scala.util.Using$.$anonfun$resources$5(Using.scala:328)
[error]     at scala.util.Using$.resource(Using.scala:261)
[error]     at scala.util.Using$.$anonfun$resources$4(Using.scala:327)
[error]     at scala.util.Using$.resource(Using.scala:261)
[error]     at scala.util.Using$.$anonfun$resources$3(Using.scala:326)
[error]     at scala.util.Using$.resource(Using.scala:261)
[error]     at scala.util.Using$.resources(Using.scala:325)
[error]     at scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.manifest classpath entry works(ZipAndJarFileLookupFactoryTest.scala:88)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]     at java.lang.reflect.Method.invoke(Method.java:566)
[error]     ...
[info] Test run scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest finished: 1 failed, 0 ignored, 2 total, 0.406s
[error] Failed: Total 8, Failed 2, Errors 0, Passed 6
[error] Failed tests:
[error] 	scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest
[error] 	scala.reflect.io.ZipArchiveTest
[error] (junit / Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
```

## Problem
The problem does not occur in some other circumstances (notably the CI). [Also @som-snytt wasn't able to reproduce it immediately.](https://github.com/scala/scala/pull/10159#issuecomment-1292015205)

I don't know exactly, what the bug is or how to trigger/prevent it, but I found that the `manifestAt` returns the wrong manifest.
Adding the following print statement yields interesting insight.
```diff
diff --git a/test/junit/scala/reflect/io/ZipArchiveTest.scala b/test/junit/scala/reflect/io/ZipArchiveTest.scala
index ec7ede4348..fd49d6ab20 100644
--- a/test/junit/scala/reflect/io/ZipArchiveTest.scala
+++ b/test/junit/scala/reflect/io/ZipArchiveTest.scala
@@ -50,7 +50,11 @@ class ZipArchiveTest {
     assertThrown[IOException](_.getMessage.contains(f.toString))(fza.iterator)
   }
 
-  private def manifestAt(location: URI): URL = ScalaClassLoader.fromURLs(List(location.toURL), null).getResource("META-INF/MANIFEST.MF");
+  private def manifestAt(location: URI): URL = {
+    val url = ScalaClassLoader.fromURLs(List(location.toURL), null).getResource("META-INF/MANIFEST.MF")
+    println(url)
+    url
+  }
 
   // ZipArchive.fromManifestURL(URL)
   @Test def `manifest resources just works`(): Unit = {
```
This prints
`jar:file:/home/anselm/Downloads/idea-IC-221.4501.155/plugins/java/lib/rt/debugger-agent.jar!/META-INF/MANIFEST.MF`
in Intellij's debugger and
`jar:file:/usr/share/java/java-atk-wrapper.jar!/META-INF/MANIFEST.MF`
in my sbt.

So, clearly, the `ScalaClassLoader.getResource` returns a manifest of a different jar than set by `fromURLs`.
This may be a bug in `ScalaClassLoader` or the classloader may be used in a wrong way in `manifestAt()`.

I hope somebody is able to reproduce this behaviour with the given hints. (Maybe add another jar with another MANIFEST to classpath or so?)

### PR DESCRIPTION
**Pull request by ansvonwa**
This does not call the parent class loader first, which may return different resources if they're on classpath.

fixes scala/bug#12677

### CODE REVIEW

**Comment by som-snytt** [comment]
Thanks!

`findResource` is made public by `java.net.URLClassLoader`.

Probably it's fine.

That line is really only constructing a URL, so even better:
```
private def manifestAt(location: Path): URL = URI.create(s"jar:file:$location!/META-INF/MANIFEST.MF").toURL
```
invoked with just `manifestAt(jar.toAbsolutePath)`. I added `toAbsolutePath` just in case. I tested under linux, I think or assume windows does the right thing.

If it were necessary to go through a class loader, the optimal would be `getResources` and take the resource which is contained by the location.

Also if it were necessary to use a class loader, it might be more natural to use `AbstractFileClassLoader` (which does not have a public `findResource`).

**Review by som-snytt — CHANGES_REQUESTED** [review]
Suggest creating the URL directly, as we have the file Path and well-known zip entry.

The proposed tweak does avoid delegating to the parent CL.

**Comment by ansvonwa** [comment]
> The proposed tweak does avoid delegating to the parent CL.

Not sure which of the solutions (yours or mine) you mean, iiuc they both don't delegate to the parent anymore.

Anyway, I think yours looks fine too, so there it is!

**Comment by som-snytt** [comment]
Thanks again, I meant your tweak, which was also clever and valid.

**Review by som-snytt — APPROVED** [review]

**Comment by som-snytt** [comment]
@SethTisue is it possible to request a windows test? I remember dotty has a special comment command.

A comment command should be called a commandment.

**Comment by SethTisue** [comment]
> is it possible to request a windows test?

I hoped this would be possible via the GitHub web UI, and.... well, I think it will be once we merge #10206 and merge it forward onto 2.13.x.

Rather than wait for that, I'm going to merge this now and if it fails on Windows we'll deal with it.

**Comment by SethTisue** [comment]
Thank you @ansvonwa and @som-snytt !

**Comment by som-snytt** [comment]
> we'll deal with it.

I'll rely on Seth's strong coping mechanisms.

**Comment by SethTisue** [comment]
It wasn't our lucky day: https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160

```
[error] Test scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.manifest classpath entry works failed: java.lang.IllegalArgumentException: Illegal character in opaque part at index 11: jar:file:C:\Users\RUNNER~1\AppData\Local\Temp\junit1086976373254541561.jar!/META-INF/MANIFEST.MF, took 0.0 sec
[3721](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3722)
[error]     at java.net.URI.create(URI.java:852)
[3722](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3723)
[error]     at scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.manifestAt$1(ZipAndJarFileLookupFactoryTest.scala:84)
[3723](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3724)
[error]     at scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.$anonfun$manifest classpath entry works$3(ZipAndJarFileLookupFactoryTest.scala:87)
[3724](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3725)
[error]     at scala.tools.nsc.classpath.ZipAndJarFileLookupFactoryTest.manifest classpath entry works(ZipAndJarFileLookupFactoryTest.scala:87)
[3725](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3726)
[error]     ...
[3726](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3727)
[error] Caused by: java.net.URISyntaxException: Illegal character in opaque part at index 11: jar:file:C:\Users\RUNNER~1\AppData\Local\Temp\junit1086976373254541561.jar!/META-INF/MANIFEST.MF
[3727](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3728)
[error]     at java.net.URI$Parser.fail(URI.java:2847)
[3728](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3729)
[error]     at java.net.URI$Parser.checkChars(URI.java:3020)
[3729](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3730)
[error]     at java.net.URI$Parser.parse(URI.java:3057)
[3730](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3731)
[error]     at java.net.URI.this(URI.java:588)
[3731](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3732)
[error]     at java.net.URI.create(URI.java:850)
[3732](https://github.com/scala/scala/actions/runs/3379006847/jobs/5609993160#step:6:3733)
[error]     ... 40 more
```

**Comment by som-snytt** [comment]
Follow-up at https://github.com/scala/scala/pull/10207

**Pull request merged**
